xarray/structure/chunks.py

```diff
@@ -83,9 +84,27 @@ def _get_chunk(var: Variable, chunks, chunkmanager: ChunkManagerEntrypoint):
         for dim, preferred_chunk_sizes in zip(dims, preferred_chunk_shape, strict=True)
     )
 
-    chunk_shape = chunkmanager.normalize_chunks(
-        chunk_shape, shape=shape, dtype=var.dtype, previous_chunks=preferred_chunk_shape
-    )
+    if _contains_cftime_datetimes(var):
+        # If we have cftime datetimes, need to preprocess them - we can't pass
+        # an object dtype into DaskManager.normalize_chunks.
+        from dask import config as dask_config
+        from dask.utils import parse_bytes
+
+        from xarray.namedarray.utils import build_chunkspec
+
+        target_chunksize = parse_bytes(dask_config.get("array.chunk-size"))
```

Comment:
How about adding `get_auto_chunk_size` to the ChunkManager class; and put the dask-specific stuff in the DaskManager.

cc @TomNicholas